#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-sysrecordings-debian.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: The system recordings path is /usr/share on debian based systems

@DPATCH@
diff -urNad freepbx-module-recordings~/functions.inc.php freepbx-module-recordings/functions.inc.php
--- freepbx-module-recordings~/functions.inc.php	2007-07-10 15:51:20.000000000 +0300
+++ freepbx-module-recordings/functions.inc.php	2007-07-10 17:17:40.000000000 +0300
@@ -2,9 +2,9 @@
 
 // Source and Destination Dirctories for recording
 global $recordings_astsnd_path; // PHP5 needs extra convincing of a global
-$recordings_save_path = "/tmp/";
 $recordings_astsnd_path = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
 $recordings_astsnd_path .= "/sounds/";
+$recordings_save_path = $recordings_astsnd_path . "/custom/";
 
 function recordings_get_config($engine) {
 	global $ext;  // is this the best way to pass this?
@@ -211,11 +211,20 @@
 
 function recordings_readdir($snddir) {
 	$files = recordings_getdir($snddir);
+	$remove_arr = array();
 	$ptr = 0;
 	foreach ($files as $fnam) {
 		$files[$ptr] = substr($fnam, strlen($snddir)+1);
+		// ignore ivr recordings
+		if (preg_match('!custom/\d+-ivrrecording!',$files[$ptr])) {
+			$remove_arr[] = $files[$ptr];
+		}
 		$ptr++;
 	}
+
+	// remove from the list custom/*-ivrrecording*
+	$files = array_diff($files , $remove_arr );
+
 	// Strip off every possible file extension
 	$flist = preg_replace("/\.(au|g723|g723sf|g726-\d\d|g729|gsm|h263|ilbc|mp3|ogg|pcm|[au]law|[au]l|mu|sln|raw|vox|WAV|wav|wav49)$/", "", $files);
 	sort($flist);
diff -urNad freepbx-module-recordings~/page.recordings.php freepbx-module-recordings/page.recordings.php
--- freepbx-module-recordings~/page.recordings.php	2007-04-02 11:21:37.000000000 +0300
+++ freepbx-module-recordings/page.recordings.php	2007-07-10 17:12:59.000000000 +0300
@@ -47,8 +47,11 @@
 		recording_sysfiles();
 		break;
 	case "newsysrec":
-		$astsnd = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
-		$astsnd .= "/sounds/";
+		// debian uses another directory for this
+		//$astsnd = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
+		//$astsnd .= "/sounds/";
+		$astsnd = "/usr/share/asterisk/sounds/";
+		
 		$sysrecs = recordings_readdir($astsnd, strlen($astsnd)+1);
 		if (recordings_add($sysrecs[$sysrec], $sysrecs[$sysrec])) {
 			$id = recordings_get_id($sysrecs[$sysrec]);
@@ -67,9 +70,9 @@
 				echo '<div class="content"><h5>'._("Failed to create").' '.$recordings_astsnd_path.'custom'.'</h5>';			
 			}		
 		} else {
-			// can't rename a file from one partition to another, must use mv or cp
-			// rename($recordings_save_path."{$dest}ivrrecording.wav",$recordings_astsnd_path."custom/{$filename}.wav");
-			exec("mv " . $recordings_save_path . "{$dest}ivrrecording.$suffix " . $recordings_astsnd_path."custom/{$filename}.$suffix");
+			// recordings are done on the same directory - with different name
+			// ivr-*.wav are generated by dialing *77
+                        rename($recordings_save_path."{$dest}ivrrecording.wav",$recordings_astsnd_path."custom/{$filename}.wav");
 			$isok = recordings_add($rname, "custom/{$filename}.$suffix");
 
 			recording_sidebar(null, $usersnum);
@@ -310,8 +313,9 @@
 }
 
 function recording_sysfiles() {
-	$astsnd = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
-	$astsnd .= "/sounds/";
+//	in debian, the sounds dir is in /usr/share/ and cannot be configured
+//	$astsnd = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
+	$astsnd = "/usr/share/asterisk/sounds/";
 	$sysrecs = recordings_readdir($astsnd, strlen($astsnd)+1);
 ?>
 	<div class="content">
@@ -337,8 +341,10 @@
 
 function recordings_display_sndfile($item, $count, $max) {
 	// Note that when using this, it needs a <table> definition around it.
-	$astsnd = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
-	$astsnd .= "/sounds/";
+	// $astsnd = isset($asterisk_conf['astvarlibdir'])?$asterisk_conf['astvarlibdir']:'/var/lib/asterisk';
+//	in debian, the sounds dir is in /usr/share/ and cannot be configured
+	$astsnd = "/usr/share/asterisk/sounds/";
+
 	$sysrecs = recordings_readdir($astsnd, strlen($astsnd)+1);
 	print "<tr><td><select name='sysrec$count'>\n";
 	echo '<option value=""'.($item == '' ? ' SELECTED' : '')."></option>\n";
